src/storage-client/src/client.rs

```diff
@@ -627,7 +627,13 @@ where
                 for id in dropped_ids {
                     let (_, shard_frontiers) = match self.uppers.get_mut(&id) {
                         Some(value) => value,
-                        None => panic!("Reference to absent collection: {id}"),
+                        None => {
+                            // Storage state can end up sending us these before we have populated
+                            // this state, because it reconciles it state as soon as we connect,
+                            // so we can just ignore `DroppedIds` for collections we are _currently
+                            // in the process of recreating_.
+                            continue;
```

Comment:
In the past not having strict checks here had hidden subtle bugs. The comment seems to imply there is something stricter we can do here instead of unconditionally continuing. Can we check if the `id` in question is in the set of "currently being created" and still panic if not?

Reply:
Hmm, I need to think about this!

Reply:
This was broken by an erroneous change added in a previous commit! Its fixed!